### DISCUSSION TITLE
Is it possible to set the proxy target dynamically?

### DISCUSSION THREAD

**Discussion by fonzane** [chimurai/http-proxy-middleware#781] [Q&A] [answered]
Hi everyone 🙂,

does someone know if it is possible to create a proxy with this middleware on the fly? I need this functionality because I want to hand over the target of the proxy request via request param.

So far, it wasn't possible for me to call the createProxyMiddleware function when the request comes in, because I couldn't manually hand over the req, res and next arguments to the createProxyMiddleware function.

Any help will be appreciated. 🙏

**Comment by chimurai** [comment, marked as answer]
Dupe of https://github.com/chimurai/http-proxy-middleware/discussions/572

 https://github.com/chimurai/http-proxy-middleware/discussions/572#discussioncomment-655288:
> Try the router option: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/router.md

**Reply by fonzane** [reply]
Yes, that works. Thank you. I can set the target according to query parameters in the request object. 

The only problem I have with this right now is that the routing only counts for one request if I set the target according to a query parameter. When the proxy transmits a webdocument, I can't proxy the following requests from the client browser to the appropriate resources.

I tried to add a temporary variable, but `console.log(this)` inside the routing function results in `undefined` (can't access a class-object on which the routing function sits on).

So instead of setting a temporary variable that will be overwritten once a new request with a specific query parameter comes in, I may need to configure the document, so it sends me the info where to proxy the requests for its resources...

Kind regards.
